### ISSUE TITLE
Formatting lost on paragraph replacement via Docs API insertText

### ISSUE THREAD

**Issue by phtevenn** [phtevenn/cv-pilot#57]
## Summary

When the native Docs API pipeline replaces paragraph text via \`deleteContentRange\` + \`insertText\`, the inserted text inherits the formatting of the paragraph's end-of-segment mark (the \`\n\` character left after deletion). That mark takes on the formatting of the **last run** in the paragraph before deletion — which is unpredictable for mixed-format paragraphs. The result is bold text becoming non-bold, or vice versa.

## Root cause

The Docs API \`insertText\` spec says:

> The text style of the inserted text is determined by the text at the insert location.

After \`deleteContentRange [start, end-1)\` the only character remaining at \`start\` is the \`\n\` paragraph end-mark. In Google Docs, this mark inherits the formatting of the **last run** of the paragraph before deletion. So:

- \`"Gate Bioscience | Scientist\tFeb 2023"\` — last run is \`"Feb 2023"\` (non-bold) → entire replacement becomes non-bold → bold on company name and title is lost
- \`"AI & ML: PyTorch, TensorFlow"\` — last run is the skills list (non-bold) → category label \`"AI & ML"\` loses its bold
- A bullet point where the last run is bold → replacement is incorrectly bolded

The outcome depends entirely on the formatting of whatever text happened to be at the end of the paragraph, making it unpredictable for any paragraph with mixed bold/non-bold runs.

## Affected paragraphs

Anything with **intra-paragraph mixed formatting**:
- Entry header lines: \`Company | Title[TAB]Date\` (bold company+title, non-bold date)
- Skills lines: \`**Label:** value\` (bold label, non-bold skills list)
- Publication author lists: \`**Author Name**, Co-author, ...\` (bold own name, non-bold others)
- Any line where a bold segment is followed by non-bold text

Uniform paragraphs (bullets, body text) are largely unaffected since all runs have the same formatting.

## What was tried

Previous DOCX round-trip approach attempted heuristics in \`docx_utils.py\`:
- Use first non-empty run's formatting
- Detect mixed-format paragraphs and fall back to \`bold=None\`
- Tab-split to apply pre/post-tab formatting separately
- \`[TAB]\` placeholder to preserve tab structure through LLM

These were fragile and incomplete. The current native Docs API approach avoids the DOCX heuristics but introduces this new formatting problem at the API level.

### Option A — Insert before delete
Instead of delete-then-insert, do insert-then-delete:
1. \`insertText\` at \`start\` — new text inherits formatting of char at \`start - 1\` (last char of previous paragraph, usually the previous paragraph's \`\n\`)
2. \`deleteContentRange [start + len(new_text), start + len(new_text) + old_length - 1)\`

Still inherits from an adjacent character but now from the *previous* paragraph's end rather than the *current* paragraph's end. More predictable in practice but still not guaranteed correct.

### Option B — Explicit \`updateTextStyle\` after each insert
After each \`insertText\`, add \`updateTextStyle\` requests to re-apply the source paragraph's formatting explicitly. Requires reading run-level formatting from the Docs API response in \`_extract_paragraphs\` and threading it through the pipeline.

For uniform paragraphs: one \`updateTextStyle\` call covering the whole paragraph.  
For mixed-format paragraphs: multiple \`updateTextStyle\` calls, one per source run — but this requires knowing how to split the replacement text across the original run boundaries (hard when LLM changes the text length).

### Option C — \`replaceAllText\` instead of delete+insert
The Docs API \`replaceAllText\` operation is documented to preserve the text style of the matched text. For paragraphs where the old text is unique in the document (true for almost all resume content), this would naturally preserve formatting without any extra style requests.

Limitations:
- Replaces all occurrences globally (risk if same phrase appears twice — unlikely for resumes)
- For mixed-format paragraphs, replacement takes formatting of the first character of the match, so the whole replacement gets that character's style (still not perfect for mixed lines)

### Option D — Run-level replacement for mixed-format paragraphs
Detect paragraphs with mixed formatting from the Docs API response. For those paragraphs, instead of replacing the whole paragraph, replace each run individually: identify which part of the LLM's replacement text corresponds to which source run (by structural markers like \`|\` separator and \`[TAB]\`), then do per-run \`deleteContentRange\` + \`insertText\` to preserve each run's formatting.

Most complex to implement but most correct. Works well for structured lines with consistent separators (e`g. \`Company | Title[TAB]Date\`). Harder for free-form mixed text.

### Option E — LLM outputs run-level structured replacements
Change the LLM output format to include formatting intent alongside the text:
\`\`\`json
{"replacements": [
  {"index": 4, "runs": [
    {"text": "Gate Bioscience", "bold": true},
    {"text": " | Scientist, Biology", "bold": true},
    {"text": "\tFeb 2023 – Present", "bold": false}
  ]}
]}
\`\`\`
The LLM sees the source paragraph's run structure (included in the prompt) and outputs replacements that match it. We then use \`updateTextStyle\` per run to re-apply formatting. Requires a richer prompt and output schema but eliminates all guesswork.

## Recommended approach

**Option C (\`replaceAllText\`) as a quick win** — low effort, fixes the majority of cases (uniform paragraphs) without any extra style logic, and naturally preserves formatting for the most common paragraph types.

**Option B or D for mixed-format paragraphs** — after Option C is in place, handle the remaining cases (entry headers, skills labels) by detecting mixed-format paragraphs from the Docs API response and either applying \`updateTextStyle\` after the insert or doing per-run replacement.

## Current pipeline reference

- \`backend/gdocs_client.py\` — \`generate_resume_native()\`, \`apply_paragraph_replacements()\`
- \`_extract_paragraphs()\` — currently extracts only text + indices, not run formatting
